File: 5차 과제/6-4/py 소스 파일/6_4.py
from header6_4 import queue
from func6_4_1 import Linked_queue

def main():
    queue=Linked_queue()
    print("*****데크 연산*****")

    queue.front_enqueue('A')
    print()
    queue.front_enqueue('B')
    print()
    queue.rear_enqueue('C')
    print()

    item=queue.front_dequeue()
    print(f"\t삭제 데이터:{item}")
    item=queue.rear_dequeue()
    print(f"\t\t삭제 데이터:{item}")

    queue.rear_enqueue('D')
    print()
    queue.front_enqueue('E')
    print()
    queue.front_enqueue('F')
    print()

if __name__ == "__main__":
    main()

File: DataStructure/5차 과제/6-4/C 소스 파일/6-4.c
#include <stdio.h>
#include "6-4.h"

queue* front = NULL;
queue* rear = NULL;

int main() {
    printf("*****데크 연산*****\n");
    front_enqueue('A');
    printf("\n");

    front_enqueue('B');
    printf("\n");

    rear_enqueue('C');
    printf("\n");

    element2 item=front_dequeue();
    printf("\t삭제 데이터:%c\n", item);

    item=rear_dequeue();
    printf("\t\t삭제 데이터:%c\n", item);

    rear_enqueue('D');
    printf("\n");

    front_enqueue('E');
    printf("\n");

    front_enqueue('F');
    printf("\n");

    printf("peek Front item:%c\n", front->data);
    printf("peek Rear item:%c\n", rear->data);

    return 0;
}
